articles/multithreading-in-java/index.md

```diff
@@ -0,0 +1,168 @@
+### What is a thread?
+
+A thread in Java is a small and light process. Many threads can run in a single process and each thread **must** be part of a process as it cannot exist on its own.
+
+### Multithreading
+
+Multithreading is the process of executing many threads concurrently. It makes sure that the CPU is utilized to the maximum. This means that one process can do many tasks at once.
+
+For example, one thread can be reading data from the database while another thread is writing the data to a file at the same time.
+
+### Multitasking
+
+Multitasking is the process of executing several tasks simultaneously. It assists to reduces the CPU idle time.
+
+Multitasking deals with multiple processes to perform multiple tasks simultaneously while multithreading deals with multiple threads in a single process to execute multiple tasks.
+
+### Benefits of Multithreading
+
+- Enables us to perform two or more tasks simultaneously.
+- Threads are independent and do not affect the performance of concurrent threads.
+- An execution of one thread will not interrupt other threads. This is useful for serving multiple clients, multiplayer games, or other mutually independent tasks.
+- We can divide a program into threads and execute them in parallel thus increasing the speed of the program execution.
+
+### Life cycle of a thread
+
+The following are the steps undergone by a thread during its execution:
+
+1. **`New`**: a thread has not started executing.
```

Comment:
Note the editing here. It will make the markdown structure look better and font in the rendered page.